### ISSUE TITLE
Struggling to capture frames from video

### ISSUE THREAD

**Issue by martinbean** [protonemedia/laravel-ffmpeg#354]
I have a video on S3. I’m trying to capture frames from this video using some [modified code from the README][1]:

```php
$opener = FFMpeg::open('videos/test.mp4');

$duration = $opener->getDurationInSeconds();
$interval = 10;

for ($seconds = 0; $seconds < $duration; $seconds = $seconds + $interval) {
    $path = sprintf('thumbnails/%05d.png', $seconds);

    $opener->getFrameFromSeconds($seconds)->export()->save($path);
}
```

So what I’m trying to do is, for every 10 seconds up until the duration of the video, capture a frame. However, this code fails with this error:

```
Illuminate\Contracts\Filesystem\FileExistsException with message 'File already exists at path: videos/test.mp4'
```

I don’t understand this error. Why is it complaining about the source video? I _know_ the source video file exists as it’s what I’m trying to capture frames from.

[1]: https://github.com/protonemedia/laravel-ffmpeg#multiple-exports-using-loops

**Comment by martinbean** [comment]
This seems to be working now. Was pretty sure I had tried assigning `$opener` back to itself. Went back to this a few days later and I’m getting frames being generated properly now. Thanks!

**Comment by martinbean** [comment]
@pascalbaljet Hey, I’ve afraid I’ve encountered this issue again.

I have a Laravel queued job called `CaptureFrames` that, well, captures frames for a video. It’s `handle` method looks like this:

```php
$opener = FFMpeg::open($this->video->getPath());

$interval = 5;
$duration = $opener->getDurationInSeconds();

if ($duration < $interval) {
    throw new RuntimeException(
        sprintf('Duration [%d] is less than interval [%d]', $duration, $interval)
    );
}

$thumbnailPath = sprintf('channels/%s/videos/%s/thumbnails', $this->video->channel->uuid, $this->video->uuid);
$thumbnailFilename = null;

for ($seconds = $interval; $seconds < $duration; $seconds = $seconds + $interval) {
    $filename = sprintf('%05d.png', $seconds);

    if ($seconds === $interval) {
        $thumbnailFilename = $filename;
    }

    $path = sprintf('%s/%s', $thumbnailPath, $filename);

    $opener = $opener->getFrameFromSeconds($seconds)->export()->save($path);
}

FFMpeg::cleanupTemporaryFiles();
```

However, when I try and manually dispatch this job, i.e.:

```php
$video = App\Media\Video::where('uuid', '=', '826c3ca7-f2dd-4412-b438-56bad3b0c251')->firstOrFail();

App\Jobs\CaptureFrames::dispatchSync($video)
```

I get an error about the _source_ file already existing:

```
Illuminate\Contracts\Filesystem\FileExistsException with message 'File already exists at path: videos/input/6182ee6821664.mp4'
```

I’m returning and re-assigning the “opener” as per your last reply. Why is it still complaining about the source file?

**Comment by martinbean** [comment]
@pascalbaljet Further, this only seems to happen if I try and use an `s3` disk. It seems to work fine with a `local` disk.

When I get this error message in the console:

```
Illuminate\Contracts\Filesystem\FileExistsException with message 'File already exists at path: videos/input/test.mp4'
```

This line is written to my application log:

```
[2021-11-04 15:48:50] local.INFO: ffmpeg running command '/var/www/html/bin/ffmpeg' '-y' '-ss' '00:00:05.00' '-i' '/tmp/0884cfe416062581/videos/input/test.mp4' '-vframes' '1' '-f' 'image2' '/tmp/0884cfe416062581/channels/7d768b0b-da0e-3193-a408-b7c4c69d0cae/videos/f5285267-7e37-3647-a0fd-5b0896ed0ad0/thumbnails/00005.png'
```

**Comment by pascalbaljet** [comment]
Fixed in v7.7.0 :)
